packages/create-project/src/helpers/samples.ts

```diff
@@ -91,13 +91,29 @@ export function hasSample(name: string): Promise<boolean> {
   );
 }
 
-export function downloadAndExtractRepo(root: string, { username, name, branch, filePath }: RepoInfo): Promise<void> {
-  return pipeline(
-    got.stream(`https://codeload.github.com/${username}/${name}/tar.gz/${branch}`),
-    tar.extract({ cwd: root, strip: filePath ? filePath.split('/').length + 1 : 1 }, [
-      `${name}-${branch}${filePath ? `/${filePath}` : ''}`,
-    ])
+export async function downloadAndExtractRepo(
+  root: string,
+  { username, name, branch, filePath }: RepoInfo
+): Promise<void> {
+  const archiveUrl = `https://codeload.github.com/${username}/${name}/tar.gz/${branch}`;
+  const archivePath = `${name}-${branch}${filePath ? `/${filePath}` : ''}`;
+
+  await pipeline(
+    got.stream(archiveUrl),
+    tar.extract({ cwd: root, strip: filePath ? filePath.split('/').length + 1 : 1 }, [archivePath])
   );
+
+  const files = await readdir(root);
+  const pipelineFailed = files.length === 0;
+  if (pipelineFailed) {
+    console.error(
+      'We were unable to download and extract the provided project.\n',
+      `Archive URL: ${archiveUrl}\n`,
+      `Archive path: ${archivePath}\n`,
+      `Sometimes this is due to the repo name changing. If that's the case, try using the new repo URL.`
```

Comment:
I don't know if I'd want to bother with stating the repo change cause, we're not expecting to change the repo name anymore 😬

Reply:
This is only when they provide their own URL, and it's the only situation I know of in which we'd reach this code path. (the `URL/package.json` is checked earlier)